hooks/price/types.go

```diff
@@ -1,10 +1,98 @@
 package price
 
-type Input struct {
+import (
+	"github.com/bandprotocol/chain/v2/pkg/obi"
+)
+
+const DefaultMultiplier = uint64(100000000)
+
+type CommonOutput struct {
+	Symbols    []string
+	Rates      []uint64
+	Multiplier uint64
+}
+
+type LegacyInput struct {
 	Symbols    []string `json:"symbols"`
 	Multiplier uint64   `json:"multiplier"`
 }
 
-type Output struct {
+type LegacyOutput struct {
 	Rates []uint64 `json:"rates"`
 }
+
+type Input struct {
+	Symbols            []string
+	MinimumSourceCount uint8
+}
+
+type Output struct {
+	Responses []Responses
+}
+
+type Responses struct {
+	Symbol       string
+	ResponseCode uint8
+	Rate         uint64
+}
+
+func MustDecodeResult(calldata, result []byte) CommonOutput {
+	var symbols []string
+	var rates []uint64
+
+	responses, err := DecodeResult(result)
```

Comment:
should DecodeResult and DecodeLegacyResult return CommonOutput directly? like we decode it to our common standard. 

think it's cleaner as this function will only just try each standard.

Reply:
I prefer this current way na

Reply:
I prefer Decode_() return a common result and return that directly